triton/config.py

```diff
@@ -12,25 +12,42 @@
 
 log = logging.getLogger(__name__)
 
-REQUIRED_CONFIG_KEYS = ['name', 'partition_key']
+REQUIRED_CONFIG_KEYS_V1 = ['name', 'partition_key']
+
+PROVIDER_GCP = 'gcp'
+PROVIDER_AWS = 'aws'
+REQUIRED_CONFIG_KEYS_V2 = ['provider']
+REQUIRED_CONFIG_VALID_PROVIDERS_V2 = [PROVIDER_AWS, PROVIDER_GCP]
+REQUIRED_CONFIG_KEYS_AWS_V2 = REQUIRED_CONFIG_KEYS_V1
+REQUIRED_CONFIG_KEYS_GCP_V2 = ['project', 'topic', 'private_key_file']
 
 _zmq_config = None
 
 
+def _validate_config(stream_name, config, required):
+    for k in required:
+        if k not in config:
+            raise errors.InvalidConfigurationError(
```

Comment:
`errors.InvalidConfigurationError` doesn't actually exist.

Reply:
Awkward as this line was taking from the previously existing code.

Added.

Reply:
I had seen this throw in the past when someone updated triton.yaml w/ invalid data. :) I had it indexed as "things to fix some day"